src/devices/Card/Ultralight/README.md

```diff
@@ -0,0 +1,159 @@
+        ultralight.Command = UltralightCommand.Read16Bytes;
+        var res = ultralight.RunUltralightCommand();
+        if (res > 0)
+        {
+            Console.Write($"  Block: {ultralight.BlockNumber:X2} - ");
+            for (int i = 0; i < 4; i++)
+            {
+                Console.Write($"{ultralight.Data![i]:X2} ");
+            }
+
+            var isReadOnly = ultralight.IsPageReadOnly(ultralight.BlockNumber);
+            Console.Write($"- Read only: {isReadOnly} ");
+
+            Console.WriteLine();
+        }
+        else
+        {
+            Console.WriteLine("Can't read card");
+            break;
+        }
+    }
+
+    Console.WriteLine("Configuration of the card");
+    // Get the Configuration
+    res = ultralight.TryGetConfiguration(out Configuration configuration);
+    if (res)
+    {
+        Console.WriteLine("  Mirror:");
+        Console.WriteLine($"    {configuration.Mirror.MirrorType}, page: {configuration.Mirror.Page}, position: {configuration.Mirror.Position}");
+        Console.WriteLine("  Authentication:");
+        Console.WriteLine($"    Page req auth: {configuration.Authentication.AuthenticationPageRequirement}, Is auth req for read and write: {configuration.Authentication.IsReadWriteAuthenticationRequired}");
+        Console.WriteLine($"    Is write lock: {configuration.Authentication.IsWrittenLocked}, Max num tries: {configuration.Authentication.MaximumNumberOfPossibleTry}");
+        Console.WriteLine("  NFC Counter:");
+        Console.WriteLine($"    Enabled: {configuration.NfcCounter.IsEnabled}, Password protected: {configuration.NfcCounter.IsPasswordProtected}");
+        Console.WriteLine($"  Is strong modulation: {configuration.IsStrongModulation}");
+    }
+    else
+    {
+        Console.WriteLine("Error getting the configuration");
+    }
+
+    NdefMessage message;
+    res = ultralight.TryReadNdefMessage(out message);
+    if (res && message.Length != 0)
+    {
+        foreach (var record in message.Records)
+        {
+            Console.WriteLine($"Record length: {record.Length}");
+            if (TextRecord.IsTextRecord(record))
+            {
+                var text = new TextRecord(record);
+                Console.WriteLine(text.Text);
+            }
+        }
+    }
+    else
+    {
+        Console.WriteLine("No NDEF message in this ");
+    }
+
+    res = ultralight.IsFormattedNdef();
+    if (!res)
+    {
+        Console.WriteLine("Card is not NDEF formated, we will try to format it");
+        res = ultralight.FormatNdef();
+        if (!res)
+        {
+            Console.WriteLine("Impossible to format in NDEF, we will still try to write NDEF content.");
+        }
+        else
+        {
+            res = ultralight.IsFormattedNdef();
+            if (res)
+            {
+                Console.WriteLine("Formating successful");
+            }
+            else
+            {
+                Console.WriteLine("Formated properly");
+            }
```

Comment:
this piece of code is very confusing IMO. can you write better Console.WriteLine messages? When can this return non-formatted ndef? what does it mean it's formatted properly vs successful?

Reply:
good catch, it's not correct!